returnn/forward.py

```diff
@@ -4,24 +4,27 @@
 
 __all__ = ["ReturnnForwardJob", "ReturnnForwardJobV2"]
 
-from sisyphus import *
-
 import copy
 import glob
 import os
 import shutil
 import subprocess
 import tempfile
-from typing import List, Optional, Union
+from typing import List, Optional, Union, get_args
+
+from sisyphus import Job, Task
+from sisyphus.delayed_ops import DelayedBase
+from sisyphus.job_path import AbstractPath
 
 from i6_core.returnn.config import ReturnnConfig
-from i6_core.returnn.training import Checkpoint as TfCheckpoint, PtCheckpoint
+from i6_core.returnn.training import Checkpoint as TfCheckpoint
+from i6_core.returnn.training import PtCheckpoint
 import i6_core.util as util
 
 
 Path = setup_path(__package__)
 
-Checkpoint = Union[TfCheckpoint, PtCheckpoint, tk.Path]
+Checkpoint = Union[TfCheckpoint, PtCheckpoint, AbstractPath, DelayedBase]
```

Comment:
```suggestion
Checkpoint = Union[TfCheckpoint, PtCheckpoint, DelayedBase]
```

`AbstractPath` is already a subclass of `DelayedBase`.

Reply:
Yes but I think it's good to highlight this anyway. Giving it a `Path` is probably way more common than any other type of `DelayedBase`.

Btw, I would use `Path` instead of `AbstractPath` though.